openmdao/docs/theory_manual/setup_stack.rst

```diff
@@ -0,0 +1,120 @@
+instantiation arguments.
+
+One question that is often asked is: why can't we just put all of our model building code into our group's
+`__init__` method so that everything is there when I instantiate the class? The answer is, when
+running a parallel model under MPI, certain systems might only be executed on certain processors.
+To save memory across the model, these systems are not fully set up on processors where they are
+not local. The only way to do this is to isolate the model building process into a custom method
+(`setup`) and only call it on the processors where you need the footprint for that group. While
+not everyone will run their models in parallel, it is a good practice to follow the stricter
+guideline so that, if someone wants to include your model in a larger parallel model, they won't
+be forced to refactor it.
+
+.. _theory_setup_vs_configure:
+
+Usage of setup vs. configure
+----------------------------
+
+The need for two methods for setting up a group arose from a need to sometimes change the linear or
+nonlinear solvers in a subgroup after it has been added. When setup is called on the `problem`, the
+setup method in each group is called recursively from top to bottom of the hierarchy. For example,
+a group may contain several components and groups. Setup is first called in that top group, during
+which, those components and groups are instantiated. However, the setup methods belonging to those sub-components
+and groups cannot be called until the top group's setup finishes. This means they are in a state where
+components and groups that are declared in the subgroup don't exist yet.
+
+To remedy this, there is a second api method called `configure` that lets you make changes to your subsystems
+after they have been created. The `configure` method is only needed with groups, and it is called
+recursively from the bottom of the hierarchy to the top, so that at any level, you can be sure that
+configure has already run for all your subsystems. This assures that changes made in higher level assemblies
+take precedence over those in lower level ones. Top precedence is given to changes made after calling setup
+on the `Problem`.
+
+Here is a quick guide covering what you can do in the setup and configure methods.
+
+**setup**
+
+ - Add subsystems
+ - Issue connections
+ - Assign linear and nonlinear solvers at this group level
+ - Change solver settings for any solver at this group level
+ - Assign Jacobians at this group level
+ - Set system execution order
+ - Add desvars, objectives, and constraints
+ - Add a case recorder to the group or a solver in that group.
+
+**configure**
+
+ - Assign linear and nonlinear solvers to subsystems
+ - Change solver settings in subsystems
+ - Assign Jacobians to subsystems
+ - Set execution order in subsystems
+ - Add a case recorder to a subsystem or a solver in a subsystem.
+
+**Things you should never do in configure**
```

Comment:
Perhaps mention why you should never do this? Consequences and/or effects might help someone recognize a problem they're having?